src/crepe/api/Vector2.h

```diff
@@ -3,29 +3,60 @@
 namespace crepe {
 
 //! 2D vector
+template <class T>
 struct Vector2 {
```

Comment:
**MENING:**

Ik zou het beter vinden als de volgende `typedef`s aangemaakt worden en gebruikt worden door de rest van de code inplaats van explicite templates (namen gestolen van [GLSL](https://www.khronos.org/opengl/wiki/Data_Type_(GLSL)#Vectors) voor bekendheid):
```cpp
typedef Vector2<int> ivec2;
typedef Vector2<unsigned int> uvec2;
typedef Vector2<float> vec2;
typedef Vector2<double> dvec2;
```

Ik heb in de code standaard nog niks gezegd over de casing van type namen, maar het huidige patroon is dat klassen allemaal CamelCase zijn en `typedef`s voor scalar snake\_case gebruiken (bijv. `game_object_id_t`). Omdat `Vector2` soortvan een utility klasse is die ervoor zorgt dat je een 'array' als scalar type kan gebruiken zou ik het goed vinden om een uitzondering te maken zodat de namen overeenkomen met die uit GLSL. Is dit een goed idee? Ik ben in ieder geval geen fan van expliciet de template parameter overal specificeren.

Reply:
Dit vind ik een goed idee. Dat maakt de code ook een stuk beter leesbaar.
Zal ik dit toevoegen in `types.h` of `Vector2.h`?

PS Waarom staat `types.h` niet in de API?
PS Als ik de typedefs toevoeg aan `types.h` en we `types.h` toevoegen aan de API, dan kan Vector2 uit de API.

Reply:
Ik had zelf trouwens ook het idee om een `typedef` te maken voor game-units en pixels. Allebei zijn gewoon int, maar dan zie je duidelijk waar je het over game-units hebt en waar je het over pixels hebt. Bijvoorbeeld:
```cpp
typedef int gu; //Game-units
typedef int px; //Pixels
```
Wellicht een slim idee om ook de volgende typedefs te maken voor Vector2:
```cpp
typedef Vector2<int> guvec2; //Game-units
typedef Vector2<int> pxvec2; //Pixels
```

Reply:
> Dit vind ik een goed idee. Dat maakt de code ook een stuk beter leesbaar. Zal ik dit toevoegen in `types.h` of `Vector2.h`?
> 
> PS Waarom staat `types.h` niet in de API? PS Als ik de typedefs toevoeg aan `types.h` en we `types.h` toevoegen aan de API, dan kan Vector2 uit de API.

Uiteindelijk krijgt de game programmeur alles in de `crepe/api` map *en* headers die vanuit alle headers in `crepe/api` worden geinclude (dwz, alle headers zodat je alles in de API kan `#include`n zonder fouten over missende headers). Een `typedef` werkt als een alias, dus ergens moet de definitie van het onderliggende type beschikbaar blijven. Helaas kunnen we de game programmeur niet verbieden om `Vector2` direct te gebruiken.

Reply:
Ik denk dat `types.h` het meest geschikt is omdat de vector typen erg generiek zijn. Ik neem aan dat de game programmeur straks `#include <crepe/api/RigidBody.h>` oid. gaat doen, en die header zal `types.h` wel moeten includen dus dat is verder geen probleem voor API gebruikers.

Reply:
Helder, ik zal het nog even aan `types.h` toevoegen